SeaBattle/Sea/Sea.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Drawing;
+namespace SeaBattle
+{
+    public class Sea
+    {
+        private List<BaseShip> Ships { get; set; }
+        public int SeaWidth;
+        public int SeaHeight;
+        public Sea(int x, int y)
+        {
+            SeaWidth= x;
+            SeaHeight= y;
+            Ships = new List<BaseShip>();
+        }
+
+        // Property to provide indexer
+        public BaseShip this[int i]
+        {
+            get
+            {
+                if (i >= 0 && i < Ships.Count)
+                    return Ships[i];
+                else
+                    throw new IndexOutOfRangeException();
+            }
+            // checking value is not null, checking value is in BaseShip family, checking Sea[i] is present
+            set
+            {
+                if (value != null)
```

Comment:
Again, it's a bad practice to have a lot of nested ifs or loops, try to simplify them or incapsulate.